DXRando/DeusEx/Classes/SmugglerElevatorTracker.uc

```diff
@@ -0,0 +1,32 @@
+class SmugglerElevatorTracker extends Trigger;
+
+function Trigger(Actor Other, Pawn Instigator)
+{
+    local DXRando dxr;
+    local int expiration;
+    local bool oldState;
+
+    if (!#defined(vanilla)) Destroy();
+
+    foreach AllActors(class'DXRando', dxr) break;
+
+    switch(dxr.localURL) {
```

Comment:
Not a huge fan of having a list of maps in this trigger class.  You should just have variables defined in the class that you can configure after spawning it (or make a static Create function that can take the parameters in).

expiration, whether it's streets or not, etc.  Keep the map data in "map scripts" like the fixup functions, and just keep the trigger itself as logic that operates on the data you pass in.

Reply:
That's originally how I implemented it, but thought it was cleaner to keep all that of those checks in one place.